### ISSUE TITLE
Requiring this package causes our app to crash

### ISSUE THREAD

**Issue by ibash** [getsentry/sentry-electron#51]
Steps:
1. ???
2. `require('@sentry/electron')`

Expected:
Works

Actual:
Causes one of our renderer processes to crash

We instantiate a webContents with `webContents.create`. If we put a preload in there that requires this package, it causes that process to crash when the app loads.

Let me know if you need any more information.

It would be great if sentry could simply give me a crash reporter url to send crashes to...

Also... https://github.com/getsentry/sentry-wizard/pull/5

**Comment by ibash** [comment]
funny enough it seems like sentry was actually able to catch the crashes it was causing... https://sentry.io/superhuman/electron/issues/498531913/

**Comment by jan-auer** [comment]
Hi @ibash, thank you for reporting this bug. Could you please clarify what you mean by "put a preload in there"?

Judging by the fact that this is a native crash, the SDK is either accessing some environment too early during WebContents creation when it's not yet allowed, or we've hit a bug in Electron. Are you using `async_hooks` or `domain` by any chance? It's also interesting, that the SDK was able to set up the CrashReporter, this already rules out a lot of possibilities.

Ideally, we could create a minimal reproducing example. Could you maybe share relevant code parts here or via email to jan.auer@sentry.io` so I can try to reproduce this locally? 

In the meanwhile, you can obtain the CrashReporter URL by using the _CSP Endpoint URL_ in _Project Settings > Client Keys (DSN)_ and replacing "csp-report" with "minidump", e.g.: `https://sentry.io/api/<id>/minidump?sentry_key=<key>`. We will show this URL in settings once we roll out full support for Minidumps.

**Comment by ibash** [comment]
I meant requiring the sentry package within a preload file (passed as an option to webContents.create).
I spent some time trying to create a minimal reproduction repo, but couldn't quite get it, so this might be something that's mostly unique to us.

In particular, this seems to be triggered by us creating an iframe in the webContents and then loading a script inside of that iframe with some code we run in the background. It's likely that there's something inside of the background code that's triggering it, but I didn't go farther than that. AFAIK we're not using `async_hooks`, and I'm fairly sure we're not using `domains` either.

Let me check what I can give you to reproduce it locally...

**Comment by ibash** [comment]
@jan-auer I can generate a build of our electron app that includes sentry (and so would reproduce the crash). If you wanted to modify the code you'd probably need to unpack / repack the asar file.

**Comment by jan-auer** [comment]
We have identified a couple of issues that cause troubles with disabled Node integration and sandbox mode. Documentation will be updated soon with current limitations and we are planning to fix those issues in the upcoming releases.

**Comment by jan-auer** [comment]
The v0.5.5 release fixes a couple of issues related to node integration and sandboxing and includes more robust internal error handling. It would be great if you could give it another try.

**Comment by jeroenvisser101** [comment]
@jan-auer we have the same error, and are using v0.5.5. It seems like that release didn't fix it.

**Comment by jeroenvisser101** [comment]
Can confirm that removing intercom, and with it all iframes, resolves the crashes too.

**Comment by jan-auer** [comment]
@jeroenvisser101 Thanks for reporting this. We have not tried out iframes yet, so there might be a bug lurking in the dark :) I will open a separate issue to track this.

**Comment by jeroenvisser101** [comment]
@jan-auer I don't think it's a separate issue, since @ibash mentions this:

> In particular, this seems to be triggered by us creating an iframe in the webContents and then loading a script inside of that iframe with some code we run in the background.

**Comment by jan-auer** [comment]
You're right. Let's rename this issue for now. 

Adding @kamilogorek @HazAT here, since they are updating the SDK currently.

**Comment by Apex2apex** [comment]
We are experiencing the same thing with our app.  We call Sentry.init in our main browser process and all the renderer processes via a preload script.  Any iFrame that got created within the renderer would crash the renderer process.

**Comment by jeroenvisser101** [comment]
@jan-auer is this still on your radar? We'd very much like to have this shipped with our initial release to aid debugging issues that might arise.

**Comment by jan-auer** [comment]
@jeroenvisser101 Yes it is, we're currently updating the Electron SDK with new internals and will look at this next.

**Comment by HazAT** [comment]
Hey, so I was playing around with this, currently working on https://github.com/getsentry/sentry-electron/pull/89 but I am not able to reproduce it.
Can you guys provide a minimal reproducible case, ideally in our example we have in the repo?!

**Comment by jeroenvisser101** [comment]
Will try to get a reproducible example today

**Comment by jeroenvisser101** [comment]
Just adding an iframe doesn't trigger anything in the application. I'll try to set up a project that similar to ours and add sentry-electron to it, but that means I'll defer from the example. Hope that's okay.

**Comment by jeroenvisser101** [comment]
@HazAT I've invited you for a private replication repo. We've ran out of seats, if needed, I'll mirror the repo on my personal account so I can add more people.

**Comment by jeroenvisser101** [comment]
Realized that we only have a public Intercom token in there, so made it public:

https://github.com/mymobia/sentry-replication-repo

**Comment by HazAT** [comment]
@jeroenvisser101 I already look at your repro case, thanks for providing it.
I did not have time yet to fully dig into it to see why it's crashing but will certainly do.

**Comment by jeroenvisser101** [comment]
I suspect this might be related: https://github.com/electron/electron/issues/12467, which means it's an upstream issue. It should be fixed in electron@3, which I'll be testing today.

**Comment by jeroenvisser101** [comment]
I can confirm using electron@3, the renderer doesn't crash and Sentry works.
